From linux-usb-devel-bounces@lists.sourceforge.net Thu Jan 25 01:33:20 2007
From: Josh Triplett <josh@freedesktop.org>
Date: Thu, 25 Jan 2007 01:32:34 -0800
Subject: USB: In init_endpoint_class, use PTR_ERR to obtain an errno value, not IS_ERR
To: linux-kernel@vger.kernel.org, linux-usb-devel@lists.sourceforge.net
Cc: Andrew Morton <akpm@osdl.org>, usb-hacking@lists.psas.pdx.edu
Message-ID: <45B87932.1070803@freedesktop.org>


init_endpoint_class calls class_create, and checks the result for an error
with IS_ERR; however, if true, it then returns the result of IS_ERR (a
boolean) rather than PTR_ERR (the actual errno).

Signed-off-by: Josh Triplett <josh@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/endpoint.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/core/endpoint.c
+++ gregkh-2.6/drivers/usb/core/endpoint.c
@@ -229,7 +229,7 @@ static int init_endpoint_class(void)
 	kref_init(&ep_class->kref);
 	ep_class->class = class_create(THIS_MODULE, "usb_endpoint");
 	if (IS_ERR(ep_class->class)) {
-		result = IS_ERR(ep_class->class);
+		result = PTR_ERR(ep_class->class);
 		goto class_create_error;
 	}
 
